common/docker-telemetry-plugin

```diff
@@ -27,40 +16,53 @@ if [ -n "$DOCKER_SOCKET" ]; then
         while ! [ -e "$DOCKER_SOCKET" ]; do
                 sleep 0.1
         done
+else
+        # docker socket not found, exiting...
+        exit
+fi
+
+# check if the file exists
+if ! [ -e "$EDITION_INFO" ]; then
+        # edition metadata not found, exiting...
+        exit
+fi
+
+# Create the engine UUID, if it doesn't exist
+# This will be eventually handled by the docker engine
+# ref. https://github.com/moby/moby/issues/33356
+if [ -f $BASE/engine_uuid ]; then
+    UUID=$(cat $BASE/engine_uuid)
+fi
+if [ -z ${UUID+x} ]; then
+    UUID=$(cat /proc/sys/kernel/random/uuid | tee $BASE/engine_uuid)
 fi
 
 # store the metrics plugin state
-plugin_name=$(docker plugin ls | grep docker/telemetry | awk 'NR==1{print $2}')
-plugin_metadata=$TAR_DIR/.plugin-metadata
+plugin_name=$(docker plugin ls | grep $PLUGIN_NAME | awk 'NR==1{print $2}')
 if docker inspect $plugin_name > /dev/null 2>&1; then
         plugin_state=$(docker plugin inspect $plugin_name | grep -o '"Enabled":.*' | sed 's/,/\\\}/g' | sed 's/ //g')
```

Comment:
To get just the `true` / `false` for `Enabled`, consider using `docker plugin inspect -f '{{.Enabled}}' $plugin_name`